src/runtime/profiler_common.cpp

```diff
@@ -350,7 +349,11 @@ WEAK void halide_profiler_stack_peak_update(void *user_context,
     // Update per-func memory stats
     for (int i = 0; i < instance->pipeline_stats->num_funcs; ++i) {
         if (f_values[i] != 0) {
-            sync_compare_max_and_swap(&(instance->funcs[i]).stack_peak, f_values[i]);
+            // On 32-bit platforms we don't want to use 64-bit
+            // atomics. Fortunately on these platforms memory usage fits into
+            // 32-bit integers.
+            sync_compare_max_and_swap((uintptr_t *)(&(instance->funcs[i]).stack_peak),
```

Comment:
This isn't safe or correct. stack_peak is a uint64, so the result will be incorrect on 32-bit platforms.

Reply:
This assumes little-endian. It will just update the low 32 bits of stack_peak. The high bits will be zero on 32-bit targets anyway.

Reply:
Wouldn't it be cleaner to just change `stack_peak` to be a uintptr_t?

Reply:
I was worried about breaking existing users of the profiler stats.